epochStart/bootstrap/fromLocalStorage.go

```diff
@@ -20,18 +20,26 @@ func (e *epochStartBootstrap) initializeFromLocalStorage() {
 	if errNotCritical != nil {
 		e.baseData.storageExists = false
 		log.Debug("no epoch db found in storage", "error", errNotCritical.Error())
-	} else {
-		e.baseData.storageExists = true
-		e.baseData.lastEpoch = latestData.Epoch
-		e.baseData.shardId = latestData.ShardID
-		e.baseData.lastRound = latestData.LastRound
-		e.baseData.epochStartRound = latestData.EpochStartRound
-		log.Debug("got last data from storage",
-			"epoch", e.baseData.lastEpoch,
-			"last round", e.baseData.lastRound,
-			"last shard ID", e.baseData.shardId,
-			"epoch start Round", e.baseData.epochStartRound)
+		return
+	}
+	if latestData.ShardID != e.destinationShardAsObserver && e.destinationShardAsObserver != core.DisabledShardIDAsObserver {
```

Comment:
will this still work if you were a validator, not an observer in the previous epoch, and the destinationShardAsObserver does not match to the shard where the validator validated previously? Will the validator be need to restart a synchronization, not start from his DB?

Reply:
This condition does not seem right. Save destinationShardAsObserver to the latestData - and return it here. That way you can make a good decision. StorageExists should be left to TRUE as the node will resolve if it is shuffled out in the function prepareEpochFromStorage

Reply:
done, removed that part of code